src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelControllerToJob2dDriverAdapter.java

```diff
@@ -0,0 +1,39 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+
+/**
+ * driver adapter to drawer with several bugs.
+ */
+public class DrawPanelControllerToJob2dDriverAdapter extends DrawPanelController implements Job2dDriver {
+	private int startX = 0, startY = 0;
+
+	public DrawPanelControllerToJob2dDriverAdapter() {
+		super();
+	}
+
+	@Override
+	public void setPosition(int x, int y) {
+		this.startX = x;
+		this.startY = y;
+	}
+
+	@Override
+	public void operateTo(int x, int y) {
+		ILine line = LineFactory.getBasicLine();
+		line.setStartCoordinates(this.startX, this.startY);
+		line.setEndCoordinates(x, y);
+		setPosition(x,y);
+
+		DrawerFeature.getDrawerController().drawLine(line);
```

Comment:
Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.